analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Branching.cs

```diff
@@ -535,5 +544,71 @@ public void Branching_NotNullConstraints_VisitsElseBranch()
                 "Else",
                 "End");
         }
+
+        [DataTestMethod]
+        [DynamicData(nameof(StringIsNullOrEmptyMethods))]
+        public void Branching_IsNullOrEmpty_SetsNullConstraintsInBranches(string methodName)
+        {
+            var code = $@"
+if (string.{methodName}(arg))
+{{
+    Tag(""If"", arg);
+}}
+else
+{{
+    Tag(""Else"", arg);
+}}
+Tag(""End"", arg);";
+            var validator = SETestContext.CreateCS(code, ", string arg").Validator;
+            validator.TagValues("If").Should().HaveCount(2)
+                .And.ContainSingle(x => x.HasConstraint(ObjectConstraint.Null))
+                .And.ContainSingle(x => x.HasConstraint(ObjectConstraint.NotNull));
+            validator.ValidateTag("Else", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
+            validator.TagValues("End").Should().HaveCount(2)
+                .And.ContainSingle(x => x.HasConstraint(ObjectConstraint.Null))
+                .And.ContainSingle(x => x.HasConstraint(ObjectConstraint.NotNull));
+        }
+
+        [DataTestMethod]
+        [DynamicData(nameof(StringIsNullOrEmptyMethods))]
+        public void Branching_IsNullOrEmpty_VisitsTrueBranchIfNull(string methodName)
+        {
+            var code = $@"
+string isNull = null;
+if (string.{methodName}(isNull))
+{{
+    Tag(""If"", isNull);
+}}
+else
+{{
+    Tag(""Else"", isNull);
+}}
+Tag(""End"", isNull);";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateTag("If", x => x.HasConstraint(ObjectConstraint.Null));
+            validator.TagValues("Else").Should().BeEmpty();
+            validator.ValidateTag("End", x => x.HasConstraint(ObjectConstraint.Null));
+        }
+
+        [DataTestMethod]
+        [DynamicData(nameof(StringIsNullOrEmptyMethods))]
+        public void Branching_IsNullOrEmpty_VisitsBothBranchesIfNotNull(string methodName)
+        {
+            var code = @$"
+string notNull = new string('c', 42);
```

Comment:
We can simplify this since #6051
```suggestion
string notNull = ""Some NotNull text"";
```